### PR TITLE
WA-66: New design for onboarding

### PR DESCRIPTION
**Pull request by fmrsabino**
Changes proposed in this pull request:
- New design for onboarding

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/test/java/pm/gnosis/heimdall/ui/onboarding/account/create/GenerateMnemonicViewModelTest.kt [hunk 1]
```diff
@@ -80,14 +80,15 @@ class GenerateMnemonicViewModelTest {
         then(accountsRepositoryMock).shouldHaveNoMoreInteractions()
         assertEquals(1, saveAccountFromMnemonicCompletable.callCount)
         assertEquals(1, saveMnemonicCompletable.callCount)
-        testObserver.assertNoErrors()
+        testObserver
+                .assertValue(DataResult(Unit))
```
Comment (rmeissner) [inline app/src/test/java/pm/gnosis/heimdall/ui/onboarding/account/create/GenerateMnemonicViewModelTest.kt]: I think there is a assertResult that does all 3 asserts in one

File: app/src/test/java/pm/gnosis/heimdall/ui/onboarding/password/PasswordSetupViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Test
+    fun setupPasswordTooShort() {
+        val observer = createObserver()
+
+        viewModel.setPassword("", "").subscribe(observer)
+
+        then(encryptionManagerMock).shouldHaveZeroInteractions()
+        observer.assertNoErrors()
+                .assertValue { it is ErrorResult && it.error is SimpleLocalizedException }
```
Comment (rmeissner) [inline app/src/test/java/pm/gnosis/heimdall/ui/onboarding/password/PasswordSetupViewModelTest.kt]: check the error message

File: app/src/test/java/pm/gnosis/heimdall/ui/onboarding/password/PasswordSetupViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Test
+    fun setupPasswordNotSame() {
+        val observer = createObserver()
+
+        viewModel.setPassword("123456", "").subscribe(observer)
+
+        then(encryptionManagerMock).shouldHaveZeroInteractions()
+        observer.assertNoErrors()
+                .assertValue { it is ErrorResult && it.error is SimpleLocalizedException }
```
Comment (rmeissner) [inline app/src/test/java/pm/gnosis/heimdall/ui/onboarding/password/PasswordSetupViewModelTest.kt]: same

File: app/src/main/java/pm/gnosis/heimdall/ui/onboarding/SetupSafeIntroActivity.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+class SetupSafeIntroActivity : BaseActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        inject()
+        setContentView(R.layout.layout_finish_setup)
+    }
+
+    override fun onStart() {
+        super.onStart()
```
Comment (rmeissner) [inline app/src/main/java/pm/gnosis/heimdall/ui/onboarding/SetupSafeIntroActivity.kt]: If you press back in this screen you will not see it again, right? Didn't we want to force interaction with either button?

Comment (fmrsabino) [reply app/src/main/java/pm/gnosis/heimdall/ui/onboarding/SetupSafeIntroActivity.kt]: I think it's unnecessary to have one more check for this in the Splash screen. If the users exit the app at this point when they return they will have the safe overview with the action to add a new safe.

File: app/src/main/java/pm/gnosis/heimdall/ui/onboarding/account/restore/RestoreAccountActivity.kt [hunk 5]
```diff
@@ -62,7 +62,7 @@ class RestoreAccountActivity : BaseActivity() {
                             onError = Timber::e)
 
     private fun updateWordCounter(mnemonic: String) {
-        val text = "${mnemonic.words().count()}/12 words"
+        val text = "${mnemonic.words().count()}/12 ${getString(R.string.words)}"
```
Comment (rmeissner) [inline app/src/main/java/pm/gnosis/heimdall/ui/onboarding/account/restore/RestoreAccountActivity.kt]: why not having the whole string as a resource? `%s/12 words`

**Pull request merged**